src/python/pants/engine/internals/graph.py

```diff
@@ -615,14 +678,20 @@ async def resolve_dependencies(
                 inference_request_type(sources_field),
             )
 
-    # Typically, dep inference will return generated subtargets. We check that their original base
-    # targets were not explicitly provided because it would be redundant to include the generated
-    # subtargets too.
-    explicitly_provided = {*provided, *itertools.chain.from_iterable(injected)}
-    remaining_inferred = {
-        dep for dep in inferred if dep.maybe_convert_to_base_target() not in explicitly_provided
+    original_addresses = {*provided.addresses, *itertools.chain.from_iterable(injected)}
+    generated_addresses = {*inferred, *itertools.chain.from_iterable(explicit_file_deps_owners)}
+    # If a generated subtarget's original base target is already included, then we leave off the
```

Comment:
Referencing the discussion from https://github.com/pantsbuild/pants/pull/10375#issuecomment-659542168 here, in case it's relevant to landing this.

Reply:
If we end up going with the suggestion in that comment, then we will need to rework multiple different call sites to stop having the explicit target supersede the generated target. So, I'm going to merge this and we could have a dedicated PR to change all the call sites in one swoop if we decide to go that route.